### DISCUSSION TITLE
using SSR within Express (examples?)

### DISCUSSION THREAD

**Discussion by sa-0001** [vikejs/vike#600] [Discussion] [answered]
in the [documentation](https://vite-plugin-ssr.com/integration#server-tools) there is the following section on how to use VPS within Express:
```
// `app` can be any server framework such as Express.js/Fastify/Koa/Hapi/...
app.get('*', async (req, res) => {
  // The `renderPage()` function, for a given URL, returns the result of our
  // `render()` hook (usually an HTML string). It doesn't know anything about Express.js and
  // can be used in any server environment, including serverless such as Cloudflare Workers
  // or Vercel.
  const pageContext = await renderPage({ urlOriginal: req.url })
  res.send(pageContext.httpResponse.body)
})
```

(i assume there is an implicit `import { renderPage } from 'vite-plugin-ssr'` at the top?)

however it's not clear how this works conceptually with `vite.config.js` and `vite build` - does the `renderPage` function automatically find the `vite.config.js` and use that?  does it always dynamically build the app, or can it use the configured `/dist` folder?  does it apply babel/minify when `NODE_ENV=production` like `vite build` would?  are all other specified plugins (such as `compression` and `visualizer`) also automatically applied?

basically it's not clear to me whether/how this works in combination with `vite build` (which i am used to using for SPA).

if someone knows of a fully-fleshed-out example repo based on express, i would be grateful to be able to study it.  (maybe i would also log a suggestion to link to it from the documentation page).

**Comment by samuelstroschein** [comment, marked as answer]
Vite build compiles the source-code. renderPage "magically" gets the compiled source code. 

The following repo uses express with vite-plugin-ssr https://github.com/inlang/inlang/tree/main/source-code/website. I am the maintainer. If you have specific questions about the codebase, I can answer them.

**Reply by sa-0001** [reply]
thank you, this is very useful (and complete!).  hopefully others will find this directly when searching.

**Comment by brillout** [comment]
https://vite-plugin-ssr.com/express

**Reply by sa-0001** [reply]
thank you for adding the page; after generating the boilerplate the usage in express is a lot more clear from reading the code.  (i don't suppose there is a direct link to the repo where the boilerplate comes from?)
